### PR TITLE
refactor: removed indirection in create_app_ctx

### LINKED ISSUE #1426: remove unnecessary wrapping in `create_app_ctx`
Currently the function returns `Result<Result<AppContext, Response<Body>>>` which can be refactored to `Result<AppContext, Response<Body>>>`

### PR DESCRIPTION
**Pull request by atamakahere-git**
**Summary:**  
Regarding the resolution of #1426.
Is this the intended change expected?

**Issue Reference(s):**  
Fixes #1426

**Build & Testing:**

- [x] I ran `cargo test` successfully.
- [x] I have run `./lint.sh --mode=fix` to fix all linting issues raised by `./lint.sh --mode=check`.

**Checklist:**

- [x] I have added relevant unit & integration tests.
- [x] I have updated the [documentation] accordingly.
- [x] I have performed a self-review of my code.
- [x] PR follows the naming convention of `<type>(<optional scope>): <title>`

[documentation]: https://github.com/tailcallhq/tailcallhq.github.io/tree/develop/docs

## Summary by CodeRabbit

- **New Features**
	- Enhanced error handling capabilities for application context creation, providing clearer distinctions between types of errors encountered.
- **Refactor**
	- Improved the structure of error handling across the application, making it more robust and user-friendly by introducing specific error variants and refining response mechanisms.

### CODE REVIEW

File: cloudflare/src/handle.rs [hunk 1]
```diff
@@ -78,7 +79,8 @@ async fn get_app_ctx(
             tracing::info!("Initialized new application context");
             Ok(Ok(app_ctx))
         }
-        Err(e) => Ok(Err(e)),
+        Err(AppCtxError::BuildError(res)) => Ok(Err(res)),
+        Err(AppCtxError::ResponseError(err)) => Err(err),
     }
 }
```
[the remark this answers is not in the document]
Comment (atamakahere-git) [reply cloudflare/src/handle.rs:86]: I suspect if this is necessary

File: cloudflare/src/handle.rs [hunk 2]
```diff
@@ -78,7 +79,8 @@ async fn get_app_ctx(
             tracing::info!("Initialized new application context");
             Ok(Ok(app_ctx))
         }
-        Err(e) => Ok(Err(e)),
+        Err(AppCtxError::BuildError(res)) => Ok(Err(res)),
+        Err(AppCtxError::ResponseError(err)) => Err(err),
```
Comment (tusharmath) [inline cloudflare/src/handle.rs:83]: We could remove the indirection in `get_app_ctx` also.

Comment (atamakahere-git) [reply cloudflare/src/handle.rs]: @tusharmath did some changes, please review.

**Comment by tusharmath** [comment]
@atamakahere-git Move it back to review, when ready :)

**Pull request closed without merging**